pkg/virtctl/expand/expand_test.go

```diff
@@ -0,0 +1,160 @@
+			vm = kubecli.NewMinimalVM(vmName)
+		})
+		It("should succeed when called on vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewVirtctlCommand(expand.COMMAND_EXPAND, vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		})
+
+		It("should succeed when called with supported output format", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewVirtctlCommand(expand.COMMAND_EXPAND, outputFormat, expand.JSON, vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		})
+
+		It("should succeed when called on vm in different namespace", func() {
+			vm.Namespace = testingNamespace
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(vm.Namespace).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewVirtctlCommand(expand.COMMAND_EXPAND, namespace, testingNamespace, vmName)
+			Expect(cmd.Execute()).To(Succeed())
+		})
+
+		It("should fail when called on non existing vm", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(nonExistingVM).Return(nil, fmt.Errorf("\"%s\" not found", nonExistingVM))
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(expand.COMMAND_EXPAND, nonExistingVM)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error expanding VirtualMachine non-existing-vm: \"non-existing-vm\" not found"))
+		})
+
+		It("should fail when called on vm that does not exist in a specified namespace", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(testingNamespace).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName).Return(nil, fmt.Errorf("\"%s\" not found", vmName))
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(expand.COMMAND_EXPAND, namespace, testingNamespace, vmName)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error expanding VirtualMachine testvm: \"testvm\" not found"))
+		})
+
+		It("should fail when called with non supported output format", func() {
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
+			vmInterface.EXPECT().GetWithExpandedSpec(vmName)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(expand.COMMAND_EXPAND, outputFormat, invalidFormat, vmName)
+			err := cmd()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("not supported output format defined: test-format"))
+		})
+	})
+
+	Context("Expand command with file input", func() {
+		BeforeEach(func() {
+			sourceDir = filepath.Join(GinkgoT().TempDir(), "sourcedir")
```

Comment:
I don't quite understand this. Do we have any `sourcedir` subdirectory hardcoded in a path? If not, then simply use the `TempDir` from ginkgo and you don't need to create the directory

Reply:
I just wanted to create a temporary file to use as input in the tests, so using `TempDir` as you suggested should be enough. I will simplify it.